### ISSUE TITLE
HydeException, traceback error on python 3.8 (mac and windows)

### ISSUE THREAD

**Issue by pyrobit** [hyde/hyde#326]
hyde
Traceback (most recent call last):
  File "/Users/brunofl/Desktop/hyde/venv/bin/hyde", line 11, in <module>
    load_entry_point('hyde==0.8.9', 'console_scripts', 'hyde')()
  File "/Users/brunofl/Desktop/hyde/venv/lib/python3.8/site-packages/pkg_resources/__init__.py", line 489, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/Users/brunofl/Desktop/hyde/venv/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2852, in load_entry_point
    return ep.load()
  File "/Users/brunofl/Desktop/hyde/venv/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2443, in load
    return self.resolve()
  File "/Users/brunofl/Desktop/hyde/venv/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2449, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/Users/brunofl/Desktop/hyde/venv/lib/python3.8/site-packages/hyde/main.py", line 6, in <module>
    from hyde.engine import Engine
  File "/Users/brunofl/Desktop/hyde/venv/lib/python3.8/site-packages/hyde/engine.py", line 5, in <module>
    from hyde.exceptions import HydeException
  File "/Users/brunofl/Desktop/hyde/venv/lib/python3.8/site-packages/hyde/exceptions.py", line 10
    raise HydeException(message), None, tb

**Comment by Descent098** [comment]
So this issue is being caused by the way the error class is created, I assume compatibility was broken along the way.

I rewrote the exception class which worked for my use case but I don't have C++ buildtools installed to run tests so your milage may vary. I rewrote https://github.com/hyde/hyde/blob/7f415402cc3e007a746eb2b5bc102281fdb415bd/hyde/exceptions.py#L4-L10

to 
```python
class HydeException(Exception):
    """Base class for exceptions from hyde."""

    def __init__(self, *args):
        if args:
            self.message = args[0]
        else:
            self.message = None
    def __repr__(self):
        """What is printed if something goes wrong"""
        if self.message:
            return "Exception raised: {}".format(self.message)
        else:
            return "Unlucky there bud"
```

and now can use it, but i'm not creating a PR because I can't run the tests to validate against.

**Comment by navilan** [comment]
@Descent098 - Thanks for the report and the fix.

**Comment by Descent098** [comment]
I am going to close this ticket as I am going to continue the fixes as part of the resolution of #270 for the 1.0.0 release
